ui/app/AppLayouts/Communities/panels/SharedAddressesAccountSelector.qml

```diff
@@ -0,0 +1,185 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+
+import StatusQ.Core 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+
+import SortFilterProxyModel 0.2
+
+import utils 1.0
+
+StatusListView {
+    id: root
+
+    property bool hasPermissions
+    property var uniquePermissionTokenKeys
+
+    // read/write properties
+    property string selectedAirdropAddress
+    property var selectedSharedAddresses: []
+
+    leftMargin: d.absLeftMargin
+    topMargin: Style.current.padding
+    rightMargin: Style.current.padding
+    bottomMargin: Style.current.padding
+
+    Instantiator {
+        id: selectionChecker
+
+        model: root.model
+
+        property var keys: []
+
+        delegate: QtObject {
+            readonly property string key: model.address
+        }
+
+        onObjectAdded: {
+            const keysCopy = [...keys]
+            keysCopy.splice(index, 0, object.key)
+            keys = keysCopy
+        }
+
+        onObjectRemoved: {
+            const keysCopy = [...keys]
+            keysCopy.splice(index, 1)
+            keys = keysCopy
+        }
+    }
+
+    Component.onCompleted: {
+        if (!root.selectedSharedAddresses.length)
+            root.selectedSharedAddresses = selectionChecker.keys // pre-select all by default
+        if (!root.selectedAirdropAddress)
+            root.selectedAirdropAddress = root.selectedSharedAddresses[0] // pick the first one
+    }
+
+    QtObject {
+        id: d
+
+        // UI
+        readonly property int absLeftMargin: 12
+
+        readonly property var airdropGroup: ButtonGroup {
+            exclusive: true
+        }
+
+        readonly property var addressesGroup: ButtonGroup {
+            exclusive: false
+        }
+
+        function firstVisibleAirdropButton() {
+            for (let i = 0; i < airdropGroup.buttons.length; i++) {
+                const button = airdropGroup.buttons[i]
+                if (button.visible)
+                    return button
+            }
+        }
```

Comment:
Yeah, will fix; almost sure it's because of this:
```qml
                    // switch to next available airdrop address when unchecking
                    if (!checked && model.address === root.selectedAirdropAddress) {
                        const firstVisibleAirdropButton = d.firstVisibleAirdropButton()
                        if (firstVisibleAirdropButton)
                            firstVisibleAirdropButton.checked = true
                    }
```

Reply:
Fixed!